packager/media/formats/mp4/mp4_media_parser.cc

```diff
@@ -410,12 +410,17 @@ bool MP4MediaParser::ParseMoov(BoxReader* reader) {
 
     // Calculate duration (based on timescale).
     int64_t duration = 0;
-    if (track->media.header.duration > 0) {
+    bool mehd_header_carryover = false;
```

Comment:
Do we need these changes here?  Can we simply check that `duration > 0` after `Rescale()`, and then call `StreamInfo::set_default_fragment_duration()` where necessary?